File: 54字符流中第一个不重复的字符.py
class Solution:
    # 返回对应char
    def __init__(self):
        self.string = ''
        self.dict = {}
        
    def FirstAppearingOnce(self):
        # write code here
        for str in self.string:
            if self.dict[str] == 1:
                return str
        return '#'
        
    def Insert(self, char):
        # write code here
        self.string += char
        if char in self.dict:
            self.dict[char] += 1
        else:
            self.dict[char] = 1

File: 54字符流中第一个不重复的字符.cpp
// Hash
class Solution
{
public:
    string str;
    char hash[256] = {0};
  //Insert one char from stringstream
    void Insert(char ch)
    {
        str += ch;
        hash[ch]++;
    }
  //return the first appearence once char in current stringstream
    char FirstAppearingOnce()
    {
        for(char ch : str) {
            if(hash[ch] == 1) return ch;
        }
        return '#';
    }
};